relay-metrics/src/statsd.rs

```diff
@@ -148,3 +148,33 @@ impl GaugeMetric for MetricGauges {
         }
     }
 }
+
+/// Returns a low-cardinality metric name for use as a tag key on statsd metrics.
+///
+/// In order to keep this low-cardinality, we only enumerate a handful of well-known, high volume
+/// names. The rest gets mapped to "other".
+pub fn metric_name_tag<S: AsRef<str>>(value: S) -> &'static str {
+    let metric_name = value.as_ref();
+    if let Some(known_name) = [
+        "c:sessions/session@none",
+        "s:sessions/user@none",
+        "s:sessions/error@none",
+        "d:transactions/duration@millisecond",
+        "s:transactions/user@none",
+        "c:transactions/count_per_root_project@none",
+    ]
+    .into_iter()
+    .find(|s| s == &metric_name)
+    {
+        return known_name;
+    }
```

Comment:
Why not just use `contains` method on slice here?
```suggestion
    let metric_name = value.as_ref();
    if [
        "c:sessions/session@none",
        "s:sessions/user@none",
        "s:sessions/error@none",
        "d:transactions/duration@millisecond",
        "s:transactions/user@none",
        "c:transactions/count_per_root_project@none",
    ].contains(&metric_name)
    {
        return metric_name;
    }
```

Reply:
you might have to look into lifetimes, but it should also work just fine.

Reply:
I first used `.contains()`, but `metric_name` does not have `'static` lifetime, which is why I return a reference to the string literal instead.